tests/csapi/apidoc_room_event_test.go

```diff
@@ -0,0 +1,206 @@
+package csapi_tests
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/matrix-org/complement/internal/match"
+)
+
+func stringSliceContains(slice []string, s string) bool {
+	for _, value := range slice {
+		if value == s {
+			return true
+		}
+	}
+	return false
+}
+
+// sytest: Events come down the correct room
+func TestEventsInCorrectRoom(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+
+	alice := deployment.Client(t, "hs1", "@alice:hs1")
+
+	const (
+		roomAmount     = 10
+		goConcurrency  = 4
+		channelTimeout = 10 * time.Second
+	)
+
+	// Create all rooms
+
+	var doneRooms = make(chan string, roomAmount)
+	var todoRooms = make(chan bool, goConcurrency)
+
+	for i := 0; i < goConcurrency; i++ {
+		go func() {
+			for {
+				_, ok := <-todoRooms
```

Comment:
There's a tidier way to do this:
```go
for _ = range todoRooms {
    // ...
}
```
This is saying "consume from `todoRooms` repeatedly until it is closed, then exit the for loop", which is what you are effectively doing here.